lib/server/services/shift/ShiftService.js

```diff
@@ -64,6 +65,29 @@ class ShiftService {
             typeSpecific: null,
         };
 
+        // Find all EoS reports from the last 16 hours and sort by most recent first
+        const pastShift = { start: new Date() - SHIFT_DURATION * 2, end: new Date() };
+        const prevShiftIssues = (await getShiftIssues(pastShift, { include: null, exclude: [] }))
+            .reverse()
+            .sort((a, b) => {
+                new Date(b.createdAt) - new Date(a.createdAt);
+            });
+
+        // Filter for the correct tags and EoS title and select the most recent report
+        const reportFormat = `End of shift report - ${shiftType} - `;
+        const tags = getEosReportTagsByType(shiftType);
+        const allEosReports = prevShiftIssues.filter((report) =>
+            tags.every((tag) => report.tags.map((x) => x.dataValues.text).includes(tag))
+            && report.title.startsWith(reportFormat));
+        const eosReport = allEosReports.length > 0 ? allEosReports[0] : { text: '' };
+
+        // Extract the 'For next shifter' information
+        const infoFromPreviousShifterRegex = /(?<=\n\n### For next shifter\n)([\S\s]*)(?=\n\n###)/g;
```

Comment:
I think this can be simplified: `/### From previous shifter\n([\s\S]*)\n\n### For next shifter/`

Reply:
It needs the positive lookahead and lookbehind so they're not included in the capture groups

Reply:
You should use the second value of match result, which is the actually matched items. Lookarounds makes the regex quite hard to read in my opinion, and I love regex...